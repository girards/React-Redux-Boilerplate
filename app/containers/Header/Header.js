import React from 'react';
import ReactRouterPropTypes from 'react-router-prop-types';
import { Menu, Image } from 'semantic-ui-react';

export default class Header extends React.PureComponent {
  render() {
    return (
      <Menu>
        <Menu.Item>
          <Image size="mini" src="https://react.semantic-ui.com/logo.png" />
        </Menu.Item>
        <Menu.Menu position="right">
          <Menu.Item
            name="login"
            onClick={() => this.props.history.push('/login')}
          />
          <Menu.Item
            name="register"
            onClick={() => this.props.history.push('/signup')}
          />
        </Menu.Menu>
      </Menu>
    );
  }
}

Header.propTypes = {
  history: ReactRouterPropTypes.history.isRequired,
};
