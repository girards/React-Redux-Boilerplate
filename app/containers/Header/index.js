/**
 *
 * Header
 *
 */
import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import { withStyles } from '@material-ui/core/styles';

import { createStructuredSelector } from 'reselect';
import makeSelectHeader from '../../redux/Header/selectors';
import Header from './Header';
import styles from './styles';

const mapStateToProps = createStructuredSelector({
  header: makeSelectHeader(),
});

function mapDispatchToProps() {
  return {};
}

export default withStyles(styles)(
  withRouter(
    connect(
      mapStateToProps,
      mapDispatchToProps,
    )(Header),
  ),
);
