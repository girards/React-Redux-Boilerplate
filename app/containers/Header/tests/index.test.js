/**
 *
 * Tests for Header
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import Header from '../Header';
import { DEFAULT_LOCALE } from '../../../i18n';

describe('<Header />', () => {
  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <Header />
      </IntlProvider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
