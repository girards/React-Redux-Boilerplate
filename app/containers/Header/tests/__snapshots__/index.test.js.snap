// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Should render and match the snapshot 1`] = `
<div
  class="ui menu"
>
  <div
    class="item"
  >
    <img
      class="ui mini image"
      src="https://react.semantic-ui.com/logo.png"
    />
  </div>
  <div
    class="right menu"
  >
    <a
      class="item"
    >
      Login
    </a>
    <a
      class="item"
    >
      Register
    </a>
  </div>
</div>
`;
